src/supla/network/html/button_update.cpp

```diff
@@ -1,26 +1,47 @@
 #include "button_update.h"
-
 #include <supla/network/web_sender.h>
 
-#ifdef ARDUINO_ARCH_ESP32
-#include <HTTPUpdateServer.h>
-HTTPUpdateServer httpUpdater;
-#else
-#include <ESP8266HTTPUpdateServer.h>
-ESP8266HTTPUpdateServer httpUpdater;
-#endif
+/*
+ Copyright (C) Petione for AC SOFTWARE SP. Z O.O.
+
+ This program is free software; you can redistribute it and/or
+ modify it under the terms of the GNU General Public License
+ as published by the Free Software Foundation; either version 2
+ of the License, or (at your option) any later version.
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+ You should have received a copy of the GNU General Public License
+ along with this program; if not, write to the Free Software
+ Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+*/
+
+// Include the full header for EspWebServer
+#include <supla/network/esp_web_server.h>
 
 using Supla::Html::ButtonUpdate;
 
 ButtonUpdate::ButtonUpdate(Supla::EspWebServer* server)
     : HtmlElement(HTML_SECTION_BUTTON_AFTER), server(server) {
-  // Automatyczna konfiguracja aktualizacji OTA
-  httpUpdater.setup(server->getServerPtr(), "/update");
+// Dynamically create the httpUpdater object
+#ifdef ARDUINO_ARCH_ESP32
+  httpUpdater = new HTTPUpdateServer();
+#else
+  httpUpdater = new ESP8266HTTPUpdateServer();
+#endif
```

Comment:
it doesn't compile on Arduino Mega target.
Arduino IDE by default compiles all .c and .cpp files - even if they are not used.
This ifdef only checks for ESP32 and for all other cases it tries to add ESP8266 update server, which doesn't exist for Arduino Mega board.

You can add
```
#ifndef ARDUINO_ARCH_AVR
```
at the top of this file and corresponding `#endif` at the end. This will exclude content of this file from being compiled on Arduino Mega board